qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp

```diff
@@ -217,6 +217,64 @@ void renameResult(WorkspaceGroup_sptr resultWorkspace,
     renameResultWithoutSpectra(resultWorkspace, fitDataBegin, fitDataEnd);
 }
 
+std::vector<std::string> strsplit(const std::string &string_in,
+                                  const std::string &delim) {
+  std::vector<std::string> out;
+  size_t last = 0;
+  size_t next = 0;
+  while ((next = string_in.find(delim, last)) != std::string::npos) {
+    out.push_back(string_in.substr(last, next - last));
+    last = next + 1;
+  }
+  out.push_back(string_in.substr(last, std::string::npos));
+  return out;
+}
+
+std::vector<std::string> srange(const std::string &string_in) {
+
+  std::vector<std::string> limits = strsplit(string_in, "-");
+  if (limits.size() == 1) {
+    return limits;
+  } else {
+    std::vector<std::string> out;
+    for (int i = boost::lexical_cast<int>(limits[0]);
+         i != boost::lexical_cast<int>(limits[1]) + 1; ++i) {
+      out.push_back(boost::lexical_cast<std::string>(i));
+    }
+    return out;
+  }
+}
+
+void renameFitWorkspace(WorkspaceGroup_sptr resultGroup,
+                        WorkspaceGroup_sptr resultWorkspace) {
+  // rename the fit data workspaces using result workspace names
+  std::size_t indexWS = 0; // index of fit workspace in resultGroup
+  auto numel = static_cast<std::size_t>(resultWorkspace->size());
+  for (std::size_t index = 0; index != numel; ++index) {
+    auto ws = resultWorkspace->getItem(index);
+    std::string name = ws->getName();
+    // extract relevent parts of the name
+    std::vector<std::string> nameParts = strsplit(name, "_iqt");
```

Comment:
```suggestion
    const auto nameParts = strsplit(name, "_iqt");
```

Making this const means you'd have to use `get` instead of `operator[]` below.

Reply:
Thanks for feedback, the last commit implements the changes you asked for. The operator [] still works when nameParts is const (so I left it like that).

Reply:
Sorry, I must have been thinking of `map` for some reason.